python/pyspark/sql/conf.py

```diff
@@ -64,6 +64,97 @@ def _checkType(self, obj, identifier):
                             (identifier, obj, type(obj).__name__))
 
 
+class ConfigEntry(object):
+    """An entry contains all meta information for a configuration"""
+
+    def __init__(self, confKey):
+        """Create a new ConfigEntry with config key"""
+        self.confKey = confKey
+        self.converter = None
+        self.default = _NoValue
+
+    def boolConf(self):
+        """Designate current config entry is boolean config"""
+        self.converter = lambda x: str(x).lower() == "true"
+        return self
+
+    def intConf(self):
+        """Designate current config entry is integer config"""
+        self.converter = lambda x: int(x)
+        return self
+
+    def stringConf(self):
+        """Designate current config entry is string config"""
+        self.converter = lambda x: str(x)
+        return self
+
+    def withDefault(self, default):
+        """Give a default value for current config entry, the default value will be set
+        to _NoValue when its absent"""
+        self.default = default
+        return self
+
+    def read(self, ctx):
+        """Read value from this config entry through sql context"""
+        return self.converter(ctx.getConf(self.confKey, self.default))
+
+
+class SQLConf(object):
+    """A class that enables the getting of SQL config parameters in pyspark"""
+
+    REPL_EAGER_EVAL_ENABLED = ConfigEntry("spark.sql.repl.eagerEval.enabled")\
```

Comment:
Can we do this by wrapping existing SQLConf? We can make them static properties by, for example, [this hack](https://github.com/graphframes/graphframes/pull/169/files#diff-e81e6b169c0aa35012a3263b2f31b330R381)

Reply:
or [this hack](https://github.com/apache/spark/blob/173fe450df203b262b58f7e71c6b52a79db95ee0/python/pyspark/ml/image.py#L37-L234)

Reply:
Great, thanks for your guidance, I'll take a look at them in detail.

Reply:
During see the currently implement, maybe we can directly use the SQLConf in SessionState? I do this in last commit 453004a. Please have a look when you have time, thanks :)